src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -7,10 +7,31 @@ package dsls.regex
  * to *remove* anything that currently appears in the file.
  */
 
+import scala.language.implicitConversions
+import scala.language.postfixOps
+
 /** The top of a class hierarchy that encodes regular expressions. */
 abstract class RegularExpression {
   /** returns true if the given string matches this regular expression */
   def matches(string: String) = RegexMatcher.matches(string, this)
+
+  def ||(other: RegularExpression) = Union(this, other)
+
+  def ~(other: RegularExpression) = Concat(this, other)
+
+  def <*> = Star(this)
+
+  /** Returns a regular expression consisting of one or more of `this`*/
+  def <+> = this ~ (this <*>)
```

Comment:
I like that you used the defined operators, like `||` and `~`. A part of me thought this would look worse for some reason than using Star, Union, or Concat, but now that I see it I think this looks a lot better.